tools/build_defs/framework.bzl

```diff
@@ -404,23 +429,29 @@ def _define_inputs(attrs):
     linking_infos_all = []
 
     bazel_headers = []
+    bazel_system_includes = []
     bazel_libs = []
 
     # This framework function-built libraries: copy result directories under
     # $EXT_BUILD_DEPS/lib-name
     ext_build_dirs = []
+    ext_build_dirs_set = {}
 
     for dep in attrs.deps:
-        provider = dep[OutputGroupInfo]
-        ext_built = provider and hasattr(provider, "gen_dir")
+        provider = dep[ExternallyBuiltTransitive]
```

Comment:
Could you give "provider" a more specific name. Both for readability and also so that you won't shadow the global built-in "provider"